server/modules/category/repository/categoryRepository.js

```diff
@@ -107,6 +108,17 @@ const fetchCategoryAndLevelsByUuid = async (surveyId, categoryUuid, draft = fals
   return R.pipe(R.values, R.head)(categories)
 }
 
+const isCategoryPublished = async (surveyId, categoryUuid, client = db) => {
```

Comment:
I wouldn't create a method just to check if it's published. you can use fetchCategoryAndLevlesByCategoryUuid and use Category.isPublished.

Reply:
In this way you avoid to fetch all the category, but I'll change it

Reply:
in this way, if we decide to change how we handle a published category, you need to update the code twice: in the Category.isPublished and in this method. it's about consistency. Fetching 1 category with levels is a very fast operation

Reply:
right, I'll change it